prime-angular-frontend/src/app/modules/enrolment/pages/care-setting/care-setting.component.ts

```diff
@@ -233,16 +221,20 @@ export class CareSettingComponent extends BaseEnrolmentProfilePage implements On
     super.nextRouteAfterSubmit(nextRoutePath);
   }
 
-  private removeIncompleteCareSettings() {
-    this.careSettings.controls
-      .forEach((control: FormGroup, index: number) => {
-        const value = control.get('careSettingCode').value;
+  private removeIncompleteCareSettings(allowEmptyCareSettings: boolean = false) {
```

Comment:
As I was mentioning you can choose whatever you want for this so you don't have to pass a value yourself and the default works for your use case.  What I provided was just an example and it means you have to pass true, but I was trying to explain you can pass nothing when invoking and have the default work for you.

Reply:
The reason I chose to pass a value is so it doesn't affect existing code that doesn't provide value, and the default value in my first pass was to run through the code as before so as not to cause any regression issues. Seeing that this method is only called once, I can revert to passing nothing and have the default value skip the code that adds one site, or we can remove that code as discussed. I am not sure if it will be needed in the future, but will keep it there just in case and have the default skip the code block at line 241.